### DISCUSSION TITLE
How to operate the `InMemoryDataset` like a list?

### DISCUSSION THREAD

**Discussion by StefanIsSmart** [pyg-team/pytorch_geometric#8976] [General]
For example, I generated an `InMemoryDataset` object, I could shuffle it by self.shuffle();
But how to remove some data like a list, for example:
If the data.name = 'Need_to_remove' (data is a Data() object), we could remove it from the created `InMemoryDataset` object.

**Comment by rusty1s** [comment]
You can do `dataset[index]` to only keep valid data objects. Alternatively, you can use the `pre_filter` argument if you want to skip certain data objects during dataset creation.
